modules/tinymce/src/core/main/ts/content/GetContentImpl.ts

```diff
@@ -22,9 +22,15 @@ const getContentFromBody = (editor: Editor, args: GetContentArgs, body: HTMLElem
   if (args.format === 'raw') {
     content = Tools.trim(TrimHtml.trimExternal(editor.serializer, body.innerHTML));
   } else if (args.format === 'text') {
-    // return empty string for text format when editor is empty to avoid bogus elements being returned in content
-    content = Zwsp.trim(body.innerText);
-    content = content === '\n' ? '' : content;
+    content = getContentFromBody(editor, { ...args, format: 'html' }, body);
+
+    if (content === '' || !Type.isString(content)) {
+      return content;
+    }
+
+    const parseBody = document.createElement('body');
```

Comment:
Interesting so sticking it into a separate body element it will render the innerText correctly on Safari? Is it that we are getting innerText on a contenteditable and it has that special rendering mode where br:s are treated differently that causes the bug?

Reply:
The issue there was that safari treated "Content in body" and "Content in a div" differently, which could cause issues with inline/not inline editors.